common/src/main/kotlin/streams/extensions/DatabaseManagementServiceExtensions.kt

```diff
@@ -1,7 +1,29 @@
 package streams.extensions
 
 import org.neo4j.dbms.api.DatabaseManagementService
+import org.neo4j.dbms.api.DatabaseNotFoundException
 import org.neo4j.kernel.internal.GraphDatabaseAPI
 import streams.utils.Neo4jUtils
 
-fun DatabaseManagementService.getSystemDb() = this.database(Neo4jUtils.SYSTEM_DATABASE_NAME) as GraphDatabaseAPI
\ No newline at end of file
+fun DatabaseManagementService.getSystemDb() = this.database(Neo4jUtils.SYSTEM_DATABASE_NAME) as GraphDatabaseAPI
+
+fun DatabaseManagementService.getDefaultDb() = this.database(Neo4jUtils.SYSTEM_DATABASE_NAME).let {
+    val defaultDbName = it.beginTx().use {
+        val columnAs = it.execute("SHOW DEFAULT DATABASE").columnAs<String>("name")
+        if (!columnAs.hasNext()) {
+            throw DatabaseNotFoundException("here is no default db into the instance")
```

Comment:
This is a critical error that should never happen but will confuse a lot of people if it does.

Change error message to "Could not detect default database in this DBMS, or none is specified.  Please check your dbms.default_database and ensure that this database is online and available"

This system command you use is right, but things I'm not sure about is if it will respond with 1 record or 3 in CC, and also what happens if the DBMS knows what the default is, but it is offline or in error condition, and so on.

Reply:
I'll remove the exception. and add the comment about `dbms.default_database` into the docs.
If it returns 3 records in CC I don't think is a problem we get the first this should be enough.
I don't think we should care about the state:
- `getDefaultDb` returns a `GraphDatabaseService` so we can use its APIs to check the status;
- `getDefaultDbName` returns a `String` so we don't care about the status;
Wdyt?

Reply:
OK I agree, we don't care about the state in particular because also the state will change a lot over the operational lifetime of the database.   But do remember that since Kafka is generating async events into Neo4j, it's pretty much assured we're going to see errors that tie to this kind of scenario.  A kafka event comes in and as far as we know we should be able to process it, but because of reconciling state, we can't or it fails.  I guess this just then takes the regular error handling / DLQ path if these kinds of things come up.

Reply:
Yes this is something that we need to manage with a hot reload function that reloads the whole plugin state.  Wdyt?

Reply:
Sounds good....but keeping in mind that if the hot reload itself runs queries on the server, it needs throttling.  Imagine lots of messages coming in, all triggering errors, all triggering hot reloads.... :)